git/tagging-versions.md

```diff
@@ -0,0 +1,115 @@
+typically tracked in Trello or another project management tool as "cards" (or
+similar). Cards have an associated identifying number. These card numbers should
+be explicitly listed in the information about a deployment as detailed below.
+
+## Versions
+
+Each deployment should have a version number associated with it. There is a
+one-to-one relationship between deployments and versions. Each version number is
+in the form `major.minor.patch`, e.g. 2.4.1, where the patch is optional and, if
+not specified, defined to be 0 (so `2.4` is a sort of shorthand for `2.4.0`).
+
+Each regularly scheduled deployment should result in a new minor version number.
+In most cases, the minor version number should be incremented. When the project
+craftsman deems the new version to be especially significant, the major version
+number should be incremented and the minor version reset to 0.
+
+A "hotfix" occurs when a bug was deployed to production or if, for some other
+reason, a change needs to be deployed outside of the regular schedule. Hotfixes
+should increment the patch number. Each new regularly scheduled deployment
+should reset the patch number to 0.
+
+## Tagging versions
+
+Git supports tags, which are names associated with a specific commit. These tags
+may be annotated with a message and sync'd between remotes (e.g. pushed up to
+`origin`).
+
+Each version should be tagged with a tag whose name is `v<version>`, e.g.
+`v2.4.1`. The tag should be annotated with a message that includes the version
+number, a summary if applicable, and a list of the card numbers and card titles
+introduced by this version. The cards should be sorted by number. Here is an
+example message for a tag named `v2.4.1`:
+
+    Version 2.4.1
+
+    Hotfix to fix a typo and fix a permission
+
+    Includes the following cards from the Trello board:
+    - #144 - Correct misspelling of "mispelled"
+    - #145 - Blacklist new spammers
+
+When creating a tag with `git tag`, use the `-a` option to create an annotated
+tag. When the tag is created, be sure to push the tag to the server with
+`git push --tags`.
+
+## Creating a deployment
+
+In most cases, you can simply tag and deploy master. But it's important to note
+that that assumes no further commits get pushed to `origin/master` during the
+deployment, because most deployment systems will get their version of `master`
+from `origin`, not from your local repository's state.
+
+Alternately, find the right commit to tag, tag it (remembering to `git fetch
+--tags` before choosing a tag name), and deploy that tag, assuming your
+deployment system allows you to specify which commit/branch/ref to deploy.
+Remember that a tag points to a commit just like a branch does, so most
+deployment systems should let you deploy the tag directly. (I know capistrano
+does.)
+
+## Creating a hotfix
+
+Hotfixes can be more tricky than a regularly scheduled deployment. Development
```

Comment:
One way to make these hotfixes easier is to follow the release based branching scheme we have https://github.com/RoleModel/BestPractices/blob/master/git/release-branching.md